flink-runtime/src/test/java/org/apache/flink/runtime/testutils/InternalMiniClusterExtension.java

```diff
@@ -109,4 +117,24 @@ public Object resolveParameter(
         }
         throw new ParameterResolutionException("Unsupported parameter");
     }
+
+    @Override
+    public void beforeEach(ExtensionContext context) throws Exception {
+        miniClusterResource.before();
+    }
+
+    @Override
+    public void afterEach(ExtensionContext context) throws Exception {
+        miniClusterResource.after();
+    }
+
+    @Override
+    public void before(ExtensionContext context) throws Exception {
+        miniClusterResource.before();
+    }
+
+    @Override
+    public void after(ExtensionContext context) throws Exception {
+        miniClusterResource.after();
+    }
```

Comment:
I saw the mini cluster resource didn't close for each test before. I guess one reason is performance.

Start a mini cluster needs some times, if we restart it frequently, test will take a long time. 

Is it necessary for this PR?

<img width="1043" alt="image" src="https://github.com/apache/flink/assets/38427477/4343a850-9472-48ea-b331-5e6fd94ae9de">